src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -869,8 +847,23 @@ private static ExtractShardId ToExtractShardId(this IMessageExtractor self)
             IMessageExtractor messageExtractor,
             ShardOptions shardOptions)
         {
-            return builder.WithShardRegion<TKey>(typeName, entityPropsFactory,
-                messageExtractor.ToExtractEntityId(), messageExtractor.ToExtractShardId(), shardOptions);
+            shardOptions.Apply(builder);
```

Comment:
had to move 1st citizen sharding code here, although it kind of duplicates the methods that work directly with the `Extractor` delegates right now.